Update api version and product version in API folder

dockerymick opened the pull request:
https://github.com/open-cluster-management/backlog/issues/11107
https://github.com/open-cluster-management/backlog/issues/11005#event-4532279481

dockerymick committed:
updating api versions

dockerymick committed:
updating the version number in api folder

nathanweatherly commented on troubleshooting/trouble_reinstall.adoc:
```diff
@@ -28,7 +28,7 @@ If you have this problem, complete the following steps:
 ACM_NAMESPACE=<namespace>
 oc delete mch --all -n $ACM_NAMESPACE
 helm ls --namespace $ACM_NAMESPACE | cut -f 1 | tail -n +2 | xargs -n 1 helm delete --namespace $ACM_NAMESPACE
-oc delete apiservice v1beta1.webhook.certmanager.k8s.io v1.admission.cluster.open-cluster-management.io v1.admission.work.open-cluster-management.io
```
Please revert this change. This particular script is for cleaning up old installs (which would reference v1beta1)

cadawson reviewed (approved):
Looks reasonable to me. Thanks!

dockerymick committed:
Update trouble_reinstall.adoc

Confirmed with Nathan that this should be reverted

dockerymick committed:
updating the apiVersion for application 10362

dockerymick committed:
Merge branch 'mj-api-11107' of github.com:open-cluster-management/rhacm-docs into mj-api-11107

dockerymick committed:
more updates

dockerymick commented on apis/application.json.adoc:
```diff
@@ -49,7 +49,7 @@ __URL__ : http://swagger.io
 === Create a application
 ....
-POST /app.k8s.io/v1beta1/namespaces/{namespace}/applications
+POST /app.k8s.io/v1beta2/namespaces/{namespace}/applications
```
@KevinFCormier Can you confirm that my update here is accurate? Should it be `v1` or `v1beta1`

KevinFCormier replied:
There is no change for this API. `v1beta1` is still correct.

dockerymick commented on apis/application.json.adoc:
```diff
@@ -128,7 +128,7 @@ __required__|Parameters describing the application to be created.|<<_rhacm-docs_
 === Query all applications
 ....
-GET /app.k8s.io/v1beta1/namespaces/{namespace}/applications
+GET /app.k8s.io/v1beta2/namespaces/{namespace}/applications
```
@KevinFCormier same question here

KevinFCormier replied:
There is no change for this API. `v1beta1` is still correct.

dockerymick commented on apis/application.json.adoc:
```diff
@@ -174,7 +174,7 @@ __required__|Namespace that you want to use, for example, default.|string
 === Query a single application
 ....
-GET /app.k8s.io/v1beta1/namespaces/{namespace}/applications/{application_name}
+GET /app.k8s.io/v1beta2/namespaces/{namespace}/applications/{application_name}
```
@KevinFCormier same question here

KevinFCormier replied:
There is no change for this API. `v1beta1` is still correct.

dockerymick commented on apis/application.json.adoc:
```diff
@@ -217,7 +217,7 @@ __required__|Namespace that you want to use, for example, default.|string
 === Delete a application
 ....
-DELETE /app.k8s.io/v1beta1/namespaces/{namespace}/applications/{application_name}
+DELETE /app.k8s.io/v1beta2/namespaces/{namespace}/applications/{application_name}
```
@KevinFCormier same question here

KevinFCormier replied:
There is no change for this API. `v1beta1` is still correct.

dockerymick commented on apis/clusterset.json.adoc:
```diff
@@ -141,7 +141,7 @@ __required__|Parameters describing the clusterset to be created.|<<_rhacm-docs_a
 ----
 {
-  "apiVersion" : "cluster.open-cluster-management.io/v1alpha1",
+  "apiVersion" : "cluster.open-cluster-management.io/v1",
```
@jamestalton Can you confirm if my update here is accurate

jamestalton replied:
i believe so

jamestalton replied:
that would be a question for foundation

dockerymick replied:
@qiujian16 can you help confirm my proposed change?

qiujian16 replied:
clusterset api is still v1alpha1

dockerymick replied:
@qiujian16 thanks for the confirmation!

dockerymick commented on apis/observability.json.adoc:
```diff
@@ -49,7 +49,7 @@ __URL__ : http://swagger.io
 === Create a multiclusterobservability resource
 ....
-POST /apis/observability.open-cluster-management.io/v1beta1/multiclusterobservabilities
+POST /apis/observability.open-cluster-management.io/v1beta2/multiclusterobservabilities
```
@clyang82 can you confirm that my change here is accurate? The update was made according to the suggestion from issue https://github.com/open-cluster-management/backlog/issues/11005#event-4532279481

dockerymick replied:
Confirmed via Slack by @marcolan018 thank you!

dockerymick commented on apis/observability.json.adoc:
```diff
@@ -98,7 +98,7 @@ __required__|Parameters describing the MultiClusterObservability resource to be
 ----
 {
-  "apiVersion": "observability.open-cluster-management.io/v1beta1",
+  "apiVersion": "observability.open-cluster-management.io/v1",
```
@clyang82 can you help confirm that this change is correct. I kept this as v1 based on your comment here: https://github.com/open-cluster-management/backlog/issues/9378#issuecomment-816408083

clyang82 replied:
Sorry for confusion. the `v1` version is for `CustomResourceDefinitions`. not for `observability`. Here the version should be `observability.open-cluster-management.io/v1beta2`

dockerymick replied:
Thanks for clarifying!

dockerymick commented on apis/policy.json.adoc:
```diff
@@ -100,7 +100,7 @@ __required__|Parameters describing the policy to be created.|<<_rhacm-docs_apis_
 ----
 {
-  "apiVersion": "policy.open-cluster-management.io/v1/v1alpha1",
+  "apiVersion": "policy.open-cluster-management.io/v1",
```
@ycao56 can you confirm that my changes in for this API is correct?

ycao56 replied:
This is correct

dockerymick commented on applications/app_sample.adoc:
```diff
@@ -36,7 +36,7 @@ The following YAML structure shows the required fields for an application and so
 [source,yaml]
 ----
-apiVersion: app.k8s.io/v1beta1
+apiVersion: app.k8s.io/v1beta2
```
@birsanv May you confirm that my changes in this file are correct?

KevinFCormier replied:
There is no change for this API. `v1beta1` is still correct.

dockerymick commented on clusters/upgrade_cluster_discon.adoc:
```diff
@@ -177,7 +177,7 @@ When you finish deploying the {cincinnati-short} instance on your hub cluster, t
 +
 ----
-apiVersion: cincinnati.openshift.io/v1beta1
+apiVersion: cincinnati.openshift.io/v1beta2
```
@jamestalton may you also confirm that this change is correct too?

jamestalton replied:
i believe so

dockerymick replied:
Thanks James!

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -116,7 +116,7 @@ For more details, see https://access.redhat.com/documentation/en-us/red_hat_open
 +
 ----
-apiVersion: observability.open-cluster-management.io/v1beta1
+apiVersion: observability.open-cluster-management.io/v1beta2
```
@clyang82 may I also receive confirmation for this change?

dockerymick replied:
@birsanv may you confirm my suggested change? I was informed that this is related to application

dockerymick replied:
Informed by @birsanv that @xiangjingli might be the better contact. Xiangjing, may you confirm my suggested update?

xiangjingli replied:
This change is related to the observability api, which is not owned by app lifecycle. @clyang82  would be the best to confirm it

clyang82 replied:
sorry for late. That is correct.  the version is  `observability.open-cluster-management.io/v1beta2`.

dockerymick replied:
No worries @clyang82 I didn't realize you all were on PTO. Thank you so much for still responding, I appreciate you

dockerymick commented on risk_compliance/gatekeeper_policy.adoc:
```diff
@@ -20,7 +20,7 @@ spec:
     - complianceType: musthave
       objectDefinition:
-        apiVersion: templates.gatekeeper.sh/v1beta1
+        apiVersion: templates.gatekeeper.sh/v1beta2
```
@ycao56 I see that v1beta1 is still used for gatekeeper policy in the policy-collection repo. Can you confirm that my change is accurate? Should this change be made at all? Should it be changed to `v1`?

ycao56 replied:
No. Please keep it as is

dockerymick committed:
update

dockerymick commented on troubleshooting/trouble_app_deploy.adoc:
```diff
@@ -11,20 +11,20 @@ If one or more of your application resources in the Subscription YAML file uses
 ----
 failed to install release: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for
-kind "Deployment" in version "extensions/v1beta1"
+kind "Deployment" in version "networking.k8s.io/v1"
```
@KevinFCormier May you confirm that my change is accurate here. I made the update based on your comment: https://github.com/open-cluster-management/backlog/issues/9378#issuecomment-807222141

KevinFCormier replied:
This doc is giving an example of an API version that is deprecated, so it should remain as-is.

Just a note that for other cases, not everything in `extensions/v1beta1` moved to `networking.k8s.io/v1`. `Deployment` actually moved to `apps/v1`. (See https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/) But we don't want to make that update here because this is specifically a troubleshooting statement about deprecated APIs.

dockerymick committed:
Update application.json.adoc

Changes based on Kevin's suggestions

dockerymick committed:
Update app_sample.adoc

dockerymick committed:
Update trouble_app_deploy.adoc

changes

dockerymick committed:
Update gatekeeper_policy.adoc

update based on Yu's comment: https://github.com/open-cluster-management/rhacm-docs/pull/2012#discussion_r621172467

jnpacker commented on services/serv_overview.adoc:
```diff
@@ -43,7 +43,7 @@ Edit the `metadata` section of your ingress service definition to add the servic
 
 ----
-apiVersion: extensions/v1beta1
+apiVersion: extensions/v1beta2
```
This should be updated to networking.k8s.io/v1 as well I believe.

dockerymick replied:
Thanks Josh!

dockerymick committed:
Update gatekeeper_policy.adoc

reverting change based on Yu's comment and confirmation

dockerymick committed:
Update trouble_app_deploy.adoc

reverting change

dockerymick committed:
Update serv_overview.adoc

Update based on Josh's comment

dockerymick committed:
Update clusterset.json.adoc

Updating API after confirmation

dockerymick committed:
Update observability.json.adoc

confirmation received from Chun: https://github.com/open-cluster-management/rhacm-docs/pull/2012#discussion_r625542934

dockerymick committed:
Updating PR to fix conflicts

dockerymick commented:
All comments and conflicts have been resolved. This PR is ready to be merged
